src/Components/Endpoints/src/Rendering/SSRRenderModeBoundary.cs

```diff
@@ -26,12 +26,13 @@ internal class SSRRenderModeBoundary : IComponent
 
     [DynamicallyAccessedMembers(Component)]
     private readonly Type _componentType;
-    private readonly IComponentRenderMode _renderMode;
     private readonly bool _prerender;
     private RenderHandle _renderHandle;
     private IReadOnlyDictionary<string, object?>? _latestParameters;
     private string? _markerKey;
 
+    public IComponentRenderMode RenderMode { get; private set; }
```

Comment:
```suggestion
    public IComponentRenderMode RenderMode { get; }
```

I don't think it gets written outside the constructor.